### PR TITLE
Added check for whether verify_function_args receives an array

### LINKED ISSUE #2670: array_key_exists PHP warning on prod site
From some recent site apache logs:

```
[Wed Dec 02 02:12:31.550577 2020] [:error] [pid 24726] [client X] PHP Warning:  array_key_exists() expects parameter 2 to be array, null given in /var/www/api/ApiResponder.php on line 1050

[Thu Dec 10 07:01:07.909623 2020] [:error] [pid 20822] [client X] PHP Warning:  array_key_exists() expects parameter 2 to be array, null given in /var/www/api/ApiResponder.php on line 1050
[Thu Dec 10 07:01:53.762494 2020] [:error] [pid 20792] [client X] PHP Warning:  array_key_exists() expects parameter 2 to be array, null given in /var/www/api/ApiResponder.php on line 1050
[Thu Dec 10 07:04:50.170886 2020] [:error] [pid 20860] [client X] PHP Warning:  array_key_exists() expects parameter 2 to be array, null given in /var/www/api/ApiResponder.php on line 1050
[Thu Dec 10 07:05:12.955053 2020] [:error] [pid 20792] [client X] PHP Warning:  array_key_exists() expects parameter 2 to be array, null given in /var/www/api/ApiResponder.php on line 1050
[Thu Dec 10 07:08:42.762976 2020] [:error] [pid 20941] [client X] PHP Warning:  array_key_exists() expects parameter 2 to be array, null given in /var/www/api/ApiResponder.php on line 1050
[Thu Dec 10 07:08:51.325137 2020] [:error] [pid 20949] [client X] PHP Warning:  array_key_exists() expects parameter 2 to be array, null given in /var/www/api/ApiResponder.php on line 1050
[Thu Dec 10 07:08:56.694774 2020] [:error] [pid 20948] [client X] PHP Warning:  array_key_exists() expects parameter 2 to be array, null given in /var/www/api/ApiResponder.php on line 1050

```

### PR DESCRIPTION
**Pull request by blackshadowshade**
Fixes #2670

### CODE REVIEW

File: src/api/ApiResponder.php [hunk 1]
```diff
@@ -1047,7 +1047,12 @@ public function process_request($args) {
      * @return string
      */
     protected function verify_function_access($args) {
-        if (array_key_exists('type', $args)) {
+        if (!is_array($args)) {
+            $result = array(
+                'ok' => FALSE,
+                'message' => '$args must be an array',
```
Comment (cgolubi1) [inline src/api/ApiResponder.php:1053]: Let's not interpolate an invalid variable that an unauthenticated caller handed us, just to be on the safe side.  :>)

I think it'd be fine just to say "Arguments to BM API functions must be arrays"

**Comment by cgolubi1** [comment]
After that's fixed: i'll do a live test just to make double-bonus sure nothing weird happens (because i think responder testing doesn't traverse this file, though maybe i'm wrong), but seems right, and thank you.

File: src/api/ApiResponder.php (hunk 1 above)
Comment (blackshadowshade) [reply src/api/ApiResponder.php:1053]: Oh, no interpolation with single quotes, so it would print the literal. However, I'll change the message anyway.

**Comment by blackshadowshade** [comment]
Message changed, and rebased.

**Comment by cgolubi1** [comment]
Thanks for this fix.

**Pull request merged**